core-web/libs/edit-content/src/lib/fields/dot-edit-content-category-field/dot-edit-content-category-field.component.ts

```diff
@@ -74,13 +90,21 @@ export class DotEditContentCategoryFieldComponent {
      *
      * @returns {void}
      */
-    showCategories(): void {
-        this.#dialogService.open(DotEditContentCategoryFieldDialogComponent, {
-            showHeader: false,
-            styleClass: 'category-field__dialog',
-            width: '1000px',
-            height: '600px',
-            position: 'center'
-        });
+    showCategoriesSidebar(): void {
+        this.disableSelectCategoriesButton.set(true);
+        this.#componentRef = this.sidebarHost.viewContainerRef.createComponent(
+            DotEditContentCategoryFieldSidebarComponent
+        );
```

Comment:
This is cool, but why? I mean, why not just add the component to the template? to avoid requests and extra loadings?

Reply:
I don't want to have a hidden component in the view that is going to handle hundreds of categories (in the worst case), that loads nodes, and in the long run that screen becomes heavy. With @defer you save the initial impact, but it doesn't eliminate it if you close it.